--- a/src/vlcal/calib/visual_camera_calibration.cpp
+++ b/src/vlcal/calib/visual_camera_calibration.cpp
@@ -36,7 +36,7 @@
   Eigen::Isometry3d T_camera_lidar = init_T_camera_lidar;
 
   // Outer loop
-  for (int i = 0; i < params.max_outer_iterations; i++) {
+  for (int i = 0; i < 1; i++) {
     Eigen::Isometry3d new_T_camera_lidar;
     switch (params.registration_type) {
       case RegistrationType::NID_BFGS:
